lib/shoryuken/polling.rb

```diff
@@ -102,5 +123,94 @@ def queue_weight(queues, queue)
         queues.count { |q| q == queue }
       end
     end
+
+    class StrictPriority < BaseStrategy
+
+      def initialize(queues)
+        # Mapping of queues to priority values
+        @queue_priorities = queues
+          .each_with_object(Hash.new(0)) { |queue, h| h[queue] += 1 }
+
+        # Priority ordering of the queues
+        @queue_order = @queue_priorities
+          .to_a
```

Comment:
I think we need to simplify this code or add a comment of the datastructure coming  out :).